src/overlays/actors/ovl_En_Zo/z_en_zo.c

```diff
@@ -79,40 +99,275 @@ static DamageTable D_8099F4E8 = {
+    Actor_SetScale(&this->actor, 0.01f);
+
+    this->actionFunc = EnZo_Walk;
+    this->actor.gravity = -4.0f;
+}
+
+void EnZo_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnZo* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void EnZo_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnZo* this = THIS;
+
+    this->actionFunc(this, globalCtx);
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 0.0f, 0.0f, 0.0f, 4);
+    EnZo_LookAtPlayer(this, globalCtx);
+    EnZo_PlayWalkingSound(this, globalCtx);
+    EnZo_UpdateCollider(this, globalCtx);
+}
+
+s32 EnZo_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* actor,
+                          Gfx** gfx) {
+    EnZo* this = (EnZo*)actor;
+
+    if (limbIndex == 15) {
+        Matrix_InsertTranslation(1500.0f, 0.0f, 0.0f, MTXMODE_APPLY);
+        Matrix_InsertXRotation_s(this->headRot.y, MTXMODE_APPLY);
+        Matrix_InsertZRotation_s(-this->headRot.x, MTXMODE_APPLY);
+        Matrix_InsertTranslation(-1500.0f, 0.0f, 0.0f, MTXMODE_APPLY);
+    }
+
+    if (limbIndex == 8) {
+        Matrix_InsertXRotation_s(-this->upperBodyRot.y, MTXMODE_APPLY);
+        Matrix_InsertZRotation_s(-this->upperBodyRot.x, MTXMODE_APPLY);
+    }
+
+    if ((limbIndex == 8) || (limbIndex == 9) || (limbIndex == 12)) {
+        rot->y += (s16)(Math_SinS(this->limbRotY[limbIndex]) * 200.0f);
+        rot->z += (s16)(Math_CosS(this->limbRotZ[limbIndex]) * 200.0f);
+    }
+    return 0;
+}
+
+void EnZo_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* actor, Gfx** gfx) {
+    EnZo* this = (EnZo*)actor;
+    Vec3f sp30 = { 400.0f, 0.0f, 0.0f };
+    Vec3f zeroVec = { 0.0f, 0.0f, 0.0f };
+
+    if (D_8099F578[limbIndex] >= 0) {
+        Matrix_MultiplyVector3fByState(&zeroVec, &this->unk_364[D_8099F578[limbIndex]]);
+    }
+    if (limbIndex == 15) {
+        Matrix_MultiplyVector3fByState(&sp30, &actor->focus.pos);
+    }
+    if (limbIndex == 4) {
+        Matrix_MultiplyVector3fByState(&zeroVec, &this->leftFootPos);
+    }
+    if (limbIndex == 7) {
+        Matrix_MultiplyVector3fByState(&zeroVec, &this->rightFootPos);
+    }
+}
+
+Gfx D_8099F5E8[] = {
+    gsDPSetRenderMode(AA_EN | Z_CMP | Z_UPD | IM_RD | CLR_ON_CVG | CVG_DST_WRAP | ZMODE_XLU | FORCE_BL |
+                          G_RM_FOG_SHADE_A,
+                      AA_EN | Z_CMP | Z_UPD | IM_RD | CLR_ON_CVG | CVG_DST_WRAP | ZMODE_XLU | FORCE_BL |
+                          GBL_c2(G_BL_CLR_IN, G_BL_A_IN, G_BL_CLR_MEM, G_BL_1MA)),
+    gsDPSetAlphaCompare(G_AC_THRESHOLD),
+    gsSPEndDisplayList(),
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zo/EnZo_Init.s")
+void EnZo_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    EnZo* this = THIS;
+    s32 i;
+    u8* alloc = GRAPH_ALLOC(globalCtx->state.gfxCtx, 0x1000);
+    u8* allocHead;
```

Comment:
Based on WIP sub_s, `alloc` is actually a dynamic shadow texture that is `64 * 64`. So could call them something like `shadowTex` and `shadowTexIter`. However if you want to wait to name them, that is also fine.

Reply:
Personally I would also prefer the `0x1000`s be replaced with `sizeof(u8) * SQ(64)`, however that is up to you.